library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -753,54 +764,154 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
             }
 
             if (setState(PINCHING)) {
-                val eps = 0.0001f
-                if (Math.abs(mAbsTargetX) < eps || Math.abs(mAbsTargetY) < eps) {
-                    // We want to interpret this as a scaled value, to work with the *actual* zoom.
-                    @ScaledPan var scaledFocusX = -detector.focusX
-                    @ScaledPan var scaledFocusY = -detector.focusY
-                    LOG.i("onScale:", "Setting focus.", "detectorFocusX:", scaledFocusX, "detectorFocusX:", scaledFocusY)
-
-                    // Account for current pan.
-                    scaledFocusX += scaledPanX
-                    scaledFocusY += scaledPanY
-
+                // We want to interpret this as a scaled value, to work with the *actual* zoom.
+                scaledFocusX = -detector.focusX
+                scaledFocusY = -detector.focusY
+                LOG.i("onScale:", "Setting focus.", "detectorFocusX:", scaledFocusX, "detectorFocusX:", scaledFocusY)
+
+                // Account for current pan.
+                scaledFocusX += scaledPanX
+                scaledFocusY += scaledPanY
+
+                mCurrentAbsTargetX = panX
+                mCurrentAbsTargetY = panY
```

Comment:
This should be nInitialAbsTargetX if I'm not mistakin

Reply:
Forget this, I was mistaking.
I though about this pan shifting while pinching again and it's now fixed.
Its much more smooth, no initial jump and works in all situations I could come up with.